### PR TITLE
Add dedicated exit code for when we find dead code

### PR DESCRIPTION
**Pull request by whosayn**
## Description

This introduces a new class vulture.utils.ExitCode, which will encode the different exit codes vulture can return.

## Related Issue

#308 

## Checklist:

- [x] I have updated the documentation in the README.md file or my changes don't require an update.
- [x] I have added an entry in CHANGELOG.md.
- [x] I have added or adapted tests to cover my changes.
- [x] I have run `tox -e fix-style` to format my code and checked the result with `tox -e style`.

### CODE REVIEW

File: tests/test_script.py [hunk 1]
```diff
@@ -4,39 +4,49 @@
 import sys
 
 from . import call_vulture, REPO, WHITELISTS
+from vulture.core import ExitCode
```
Comment (jendrikseipp) [inline tests/test_script.py:7]:
```suggestion
from vulture.utils import ExitCode
```

File: vulture/core.py [hunk 2]
```diff
@@ -217,7 +218,7 @@ def get_list(typ):
 
         self.filename = Path()
         self.code = []
-        self.found_dead_code_or_error = False
+        self.found_dead_code_or_error = ExitCode.NoDeadCode
```
Comment (jendrikseipp) [inline vulture/core.py:221]:
```suggestion
        self.exit_code = ExitCode.NoDeadCode
```
Let's rename the variable.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 * Add `UnicodeEncodeError` exception handling to `core.py` (milanbalazs, #299).
 * Add whitelist for `Enum` attributes `_name_` and `_value_` (Eugene Toder, #305).
 * Run tests and add PyPI trove for Python 3.11 (Jendrik Seipp).
+* Add dedicated exit code for when we find dead code
```
Comment (jendrikseipp) [inline CHANGELOG.md:6]: Start a new release section, and add your name or handle if you like:

```suggestion
# 2.9 (unreleased)
* Use exit code 3 when dead code is found (whosayn, #319).
```

**Review by jendrikseipp — CHANGES_REQUESTED** [review]
Thanks for the PR! I like your enum refactoring! Once you've taken care of my small nitpicks, this can be merged.

File: vulture/core.py (hunk 2 above)
Comment (whosayn) [reply vulture/core.py]: I made the suggested changes. thanks!

**Comment by jendrikseipp** [comment]
Thanks!

**Pull request merged**